lib/backend/lite/lite.go

```diff
@@ -887,6 +785,92 @@ func (l *Backend) DeleteRange(ctx context.Context, startKey, endKey []byte) erro
 	})
 }
 
+func (l *Backend) ConditionalUpdate(ctx context.Context, i backend.Item) (*backend.Lease, error) {
+	if i.Key == nil {
+		return nil, trace.BadParameter("missing parameter key")
+	}
+
+	rev := backend.CreateRevision()
+	err := l.inTransaction(ctx, func(tx *sql.Tx) error {
+		now := l.clock.Now().UTC()
+
+		stmt, err := tx.PrepareContext(ctx, "UPDATE kv SET value = ?, expires = ?, modified = ?, revision = ? WHERE key = ? AND revision = ?")
```

Comment:
We should check that the existing value isn't expired here; the current write operations (other than `Update`) do the same.